spec/functional/resource/powershell_spec.rb

```diff
@@ -56,6 +56,21 @@
       resource.run_action(:run)
     end
 
+    it "returns the -27 for a powershell script that exits with -27" do
+      file = Tempfile.new(['foo', '.ps1'])
+      begin
+        file.write "exit -27"
+        file.close
+        resource.code(". \"#{file.path}\"")
```

Comment:
Would you need to test this as "& \"#{file.path}\"" instead of . to force it to run as an external command instead of getting sourced?